File: pertemuan12/Prak/PRAK12_2473021_PY/B_Segitiga.py
"""
Pertemuan 12
[PRAKTIKUM] sistem yang menerima input panjang sisi a, panjang sisi b, dan sudut b dari sebuah segitiga.

@author 2473021-Febrianus Leona Putra
@version 10 Desember 2024
"""
import math

def hitungA(a, b, sudut_B):
    sudut_B_Radian = math.radians(sudut_B)
    sudut_A_Radian = math.asin(math.sin(sudut_B_Radian) * a/b)
    sudut_A = math.degrees(sudut_A_Radian)
    return sudut_A

print("Sudut dan panjang sisi segitiga")
a = int(input("Masukkan panjang sisi a: "))
b = int(input("Masukkan panjang sisi b: "))
sudut_B = int(input("Masukkan sudut b: "))
sudut_A = hitungA(a, b, sudut_B)

print("\nDalam segitiga ABC, jika")
print("(a =", a,"),", sep="")
print("(b =", b,"),", sep="")
print("dan (C =", sudut_B, "^\circ),", sep="")
print("cari sudut (A) menggunakan hukum sin!")
print(f"\nSudut (A): {sudut_A:.2f}")

File: pertemuan12/Prak/PRAK12_2473021_JAVA/B_Segitiga.java
/**
 * Pertemuan 12
 * [PRAKTIKUM] sistem yang menerima input panjang sisi a, panjang sisi b, dan sudut b dari sebuah segitiga.
 *
 * @author 2473021-Febrianus Leona Putra
 * @version 10 Desember 2024
 */
package pertemuan12.Prak.PRAK12_2473021_JAVA;

import java.util.Scanner;

public class B_Segitiga {

    public static double hitungA(double a, double b, double sudutB) {
        double sudutBRadian = Math.toRadians(sudutB);
        double sudutARadian = Math.asin(Math.sin(sudutBRadian) * a / b);
        return Math.toDegrees(sudutARadian);
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.println("Sudut dan panjang sisi segitiga");

        System.out.print("Masukkan panjang sisi a: ");
        double a = input.nextDouble();

        System.out.print("Masukkan panjang sisi b: ");
        double b = input.nextDouble();

        System.out.print("Masukkan sudut b: ");
        double sudutB = input.nextDouble();

        double sudutA = hitungA(a, b, sudutB);

        System.out.println("\nDalam segitiga ABC, jika");
        System.out.println("(a = " + a + "),");
        System.out.println("(b = " + b + "),");
        System.out.println("dan (C = " + sudutB + "^\\circ),");
        System.out.println("cari sudut (A) menggunakan hukum sin!");
        System.out.printf("\nSudut (A): %.2f%n", sudutA);

        input.close();
    }
}
